Stops machine frames and circuitboards referring to manipulators as micro-manipulators

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
![](https://i.imgur.com/hFaW4Kf.png)
1. Gives stock parts a base_name variable and sets it to manipulator for micro-manipulators
2. Reorders sprites of stock parts in stock_parts.dmi to be together and in order of tier. [Image](https://i.imgur.com/RuIU2oP.png)
3. Makes component part lists on machine frames and circuitboards nicer and makes them identical between the two
4. Improves code for generating these lists
5. Adds numtotext() proc which these lists use




<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

1. Solves the issue where examining machines says they require a micro-manipulator when any manipulator will do
2. Makes sprites easier to find
3. Wows players by using actual words for numbers in generated parts lists
4. Cleans up ancient code
5. This proc can be used for other things, e.g. station names (which I have now incorporated it into)

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: Machine frames and circuit boards no longer call manipulators micro-manipulators when examined
code: Improves code in constructable_frame.dm and circuitboard.dm
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Renames micro-manipulator to manipulator

stock_parts_designs.dm
- Changes design name to Basic Manipulator
- Changes id to basic_manipulator

stock_parts.dm
- Changes name to manipulator
- Changes desc to specify that it's still a micro-manipulator
- Changes icon_state to "mani"

all_nodes.dm
- Changes its node name to basic_manipulator

stock_parts.dmi
- Renames icon in .dmi
- Reorders icons in .dmi so stock parts are together and in order from worst to best

constructable_frame.dm
- Clarifies a couple of var names in examine for machine frames
- Gives requires string info span to differentiate it from description

SuperNovaa41 commented:
Soul removal

Mothblocks commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -14,6 +14,7 @@
 		"basic_matter_bin",
 		"basic_micro_laser",
 		"basic_scanning",
+		"basic_manipulator",
```
ABC order

cacogen replied:
Oh yeah I was gonna correct that but then I forgot

Mothblocks replied:
It's mostly my fault for not setting up linters on this yet 😅

cacogen committed:
Puts it in the correct alphabet slot

Farquaar commented:
Any plans to change micro-laser as well?

cacogen commented:
Well no because every tier is a micro-laser and what changes is the power. Although maybe I could fix manipulators by similarly making every tier a micro-manipulator instead.

![image](https://user-images.githubusercontent.com/25089914/153731704-2b08b560-be2e-4858-8a0d-d772ddffd785.png)
![](https://i.imgur.com/GC0DcqQ.png)

newfren commented:
the phrase "femto micro" is very silly to me. you could just make machines request "manipulators" instead

cacogen commented:
My argument that it doesn't matter even if the prefix is supposed to refer to size is:

1) They aren't microscopic (or nanoscopic)
2) They're all the same size
2) There's no such thing as picoscopic or femtoscopic
3) They all have the same description that only refers to them as "tiny little"

cacogen commented:
> A micromanipulator is a device which is used to physically interact with a sample under a microscope.

So they're all micro manipulators, objects capable of manipulating microscopic things. The higher tiers of micro manipulator are just capable of manipulating even smaller things. 

You could shorten them to things like femto-manipulator the way we do at the moment, but that means the base object is considered a micro-manipulator and the other tiers aren't. 

That causes the problem when examining machine frames where it asks for a micro-manipulator when any tier of manipulator will do. I first solved the problem by changing the base micro-manipulator to just manipulator, but I don't like that solution either.

I'll probably add a base name string to stock parts that machine frames can pull from instead if it exists.

cacogen committed:
This took forever

cacogen committed:
Resouls the micro-manipulator desc

Fikou commented on code/modules/research/designs/stock_parts_designs.dm:
```diff
@@ -113,10 +113,10 @@
-/datum/design/micro_mani
-	name = "Micro Manipulator"
+/datum/design/basic_manipulator
+	name = "Basic Manipulator"
 	desc = "A stock part used in the construction of various devices."
-	id = "micro_mani"
+	id = "basic_manipulator"
```
you sure you still want this change in if you took a different solution?

cacogen replied:
Yeah, it puts them next to the other tier 1 parts in the stock part list and names them similarly, which helps new players understand they're tier 1.

cacogen commented:
pls merge

cacogen committed:
I forgot about the \s macro what a dumb fuck

cacogen committed:
Num to text

cacogen committed:
Undoes some changes I don't care about

AlinaStarkova commented:
"Wows players by using actual words for numbers in generated parts lists"

not sure about this one, numbers as words are harder to notice, and for those who arent fluent in english theres no reason to force the use of words to see the numbers, most games that denote costs usually have the word and number together, like "one (1) manipulator"

admittedly it really doesnt matter since nobody actually reads that, as they just fill a BRPD and just spam machines with it

cacogen commented:
I meant that in kind of an ironic way. I still think it looks good though.

The game uses words as numbers elsewhere so this feels more like concern trolling because you didn't like something I wrote more than anything.

I can't speak for other players but I use these all the time, particularly if I don't want to waste mats printing a bunch of parts. But the scenario you presented doesn't even make sense because you'd need to check the machine for things like glass sheets or cable coils. Would you prefer they were just removed and people had to consult the wiki again?

AlinaStarkova commented:
> I meant that in kind of an ironic way. I still think it looks good though.
> 
> The game uses words as numbers elsewhere so this feels more like concern trolling because you didn't like something I wrote more than anything.
> 
> I can't speak for other players but I use these all the time, particularly if I don't want to waste mats printing a bunch of parts. But the scenario you presented doesn't even make sense because you'd need to check the machine for things like glass sheets or cable coils. Would you prefer they were just removed and people had to consult the wiki again? 

your being a bit of an ass just for me bringing up a legitimate concern. not everyone who plays is native English speakers and doing "one (1)" is a pretty standard accessibility thing most games do, i dont think thats concern trolling to mention. and yes actually, I do think the other areas of the game that only use wordage for numbers hurt accessibility. no, I wouldn't prefer it be removed entirely, i never even suggested anywhere close to such, no need for sarcasm. 

as for glass sheets or cable coils, sure, thats checked if your confused why you cant finish the machine, but lets be real, you bped the parts in, screwdriver, if it doesnt work you slap glass and cable at it and try again. if it still doesnt work then you check whats missing. thats basically how most engineers build machines.

Ryll-Ryll commented on code/__HELPERS/text.dm:
```diff
@@ -809,6 +809,31 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 		else
 			return "[number]\th"
 
+/proc/numtotext(number)
```
Should specify either in the name (inttotext?) or the autodoc that this only supports integers

Ryll-Ryll commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -31,39 +31,38 @@
+		var/list/nice_list = list()
+		for(var/atom/component as anything in req_components)
+			if(!ispath(component))
+				continue
```
Should probably throw a stack_trace() here so we know what's wrong

cacogen committed:
Makes convert_integer_to_words() proc very robust indeed

- Adds it to station names
- Cleans up my machine frame and circuitboard code a bit
- Incorporate Shaps' requested changes

cacogen committed:
when it gets loudly it gets very loud indeed
